web/html/src/components/states-picker.tsx

```diff
@@ -0,0 +1,447 @@
+        this.setMessages(MessagesUtils.error(t("An error occurred on save.")));
+      }
+    );
+    return request;
+  };
+
+  onSearchChange = (event) => {
+    this.setState({
+      filter: event.target.value,
+    });
+  };
+
+  getSortedList = (data) => {
+    const [assigned, unassigned] = _partition(data, (d) => d.assigned);
+    return _sortBy(assigned, "position").concat(_sortBy(unassigned, (n) => n.name.toLowerCase()));
+  };
+
+  search = () => {
+    return Promise.resolve().then(() => {
+      if (this.state.filter !== this.state.search.filter) {
+        // Since we don't commit our changes to the backend in case of state type we perform a local search
+        this.props.type === "state"
+          ? this.stateTypeSearch()
+          : Network.get(this.props.matchUrl(this.state.filter)).then((data) => {
+              this.setState({
+                search: {
+                  filter: this.state.filter,
+                  results: this.getSortedList(data),
+                },
+              });
+              this.clearMessages();
+            });
+      }
+    });
+  };
+
+  stateTypeSearch = () => {
+    this.setState({
+      search: {
+        filter: this.state.filter,
+        results: this.state.channels.filter((c) => c.name.includes(this.state.filter)),
+      },
+    });
+    this.clearMessages();
+  };
+
+  addChanged = (original, key, selected) => {
+    const currentChannel = this.state.changed.get(key);
+    if (
+      !DEPRECATED_unsafeEquals(currentChannel, undefined) &&
+      DEPRECATED_unsafeEquals(selected, currentChannel.original.assigned)
+    ) {
+      this.state.changed.delete(key);
+    } else {
+      this.state.changed.set(key, {
+        original: original,
+        value: Object.assign({}, original, { assigned: selected }),
+      });
+    }
+    this.setState({
+      changed: this.state.changed,
+    });
+  };
+
+  handleSelectionChange = (original) => {
+    return (event) => {
+      this.addChanged(original, event.target.value, event.target.checked);
+    };
+  };
+
+  tableBody = () => {
+    const elements: React.ReactNode[] = [];
+    let rows: any[] = [];
+    rows = this.state.search.results.map((channel) => {
+      const changed = this.state.changed.get(channelKey(channel));
+      if (changed !== undefined) {
+        return changed;
+      } else {
+        return {
+          original: channel,
+        };
+      }
+    });
+
+    for (var row of rows) {
+      const changed = row.value;
+      const currentChannel = changed === undefined ? row.original : changed;
+
+      elements.push(
+        <tr
+          id={currentChannel.label + "-row"}
+          key={currentChannel.label}
+          className={changed !== undefined ? "changed" : ""}
+        >
+          <td>
+            {channelIcon(currentChannel)}
+            {currentChannel.type !== "internal_state" ? (
+              /* eslint-disable-next-line jsx-a11y/anchor-is-valid */
```

Comment:
Instead of `<a href="#">`, we should be able to use something along the lines of `<button className="is-plain">` if I remember correctly.